app/helpers/sessions_helper.rb

```diff
@@ -3,16 +3,37 @@ def log_in user
     session[:user_id] = user.id
   end
 
+  def remember user
+    user.remember
+    cookies.permanent.signed[:user_id] = user.id
```

Comment:
`cookies.permanent.signed` dùng signed với `cookies.permanent[:remember_token] ` ko dùng signed có gì khác nhau e? vào tại sao chỗ này e cần dùng signed ?

Reply:
Theo em hiểu signed dùng để mã hóa, vì id là giá  trị cố định nên để đảm bảo an toàn mình cần phải mã hóa nó khi lưu vào cookie, còn token là một một chuỗi ngẫu nhiên nên mình không cần mã hóa ạ.